### DISCUSSION TITLE
why my project pnpm install fail，thanks

### DISCUSSION THREAD

**Discussion by fuj0115** [elizaOS/eliza#987] [Q&A]
PS D:\develop\eliza> pnpm install
Scope: all 35 workspace projects
Lockfile is up to date, resolution step is skipped
Packages: +3703
++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++  
Progress: resolved 0, reused 3232, downloaded 0, added 3703, done
node_modules/canvas: Running install script...
node_modules/puppeteer: Running postinstall script...
node_modules/@discordjs/opus: Running install script, failed in 7.8s
node_modules/@discordjs/opus install$ node-pre-gyp install --fallback-to-build
│ node-pre-gyp info it worked if it ends with ok
│ node-pre-gyp info using node-pre-gyp@0.4.5
│ node-pre-gyp info using node@23.3.0 | win32 | x64
│ (node:2488) [DEP0040] DeprecationWarning: The `punycode` module is deprecated. Please use a userland alternative instead.
│ (Use `node --trace-deprecation ...` to show where the warning was created)
│ node-pre-gyp info check checked for "D:\develop\eliza\node_modules\@discordjs\opus\prebuild\node-v131-napi-v3-win32-x64-unknown-unknown\opus.node" (not found)
│ node-pre-gyp http GET https://github.com/discordjs/opus/releases/download/v0.9.0/opus-v0.9.0-node-v131-napi-v3-win32-x64-unknown-unknown.tar.gz
│ node-pre-gyp ERR! install response status 404 Not Found on https://github.com/discordjs/opus/releases/download/v0.9.0/opus-v0.9.0-node-v131-napi-v3-win32-x64-unknown-unknown.tar.gz 
│ node-pre-gyp WARN Pre-built binaries not installable for @discordjs/opus@0.9.0 and node@23.3.0 (node-v131 ABI, unknown) (falling back to source compile with node-gyp) 
│ node-pre-gyp WARN Hit error response status 404 Not Found on https://github.com/discordjs/opus/releases/download/v0.9.0/opus-v0.9.0-node-v131-napi-v3-win32-x64-unknown-unknown.tar.gz 
│ gyp info it worked if it ends with ok
│ gyp info using node-gyp@10.2.0
│ gyp info using node@23.3.0 | win32 | x64
│ gyp info ok 
│ gyp info it worked if it ends with ok
│ gyp info using node-gyp@10.2.0
│ gyp info using node@23.3.0 | win32 | x64
│ gyp info find Python using Python version 3.12.8 found at "D:\devlopTools\Python\Python312\python.exe"
│ gyp http GET https://nodejs.org/download/release/v23.3.0/node-v23.3.0-headers.tar.gz
│ gyp http 200 https://nodejs.org/download/release/v23.3.0/node-v23.3.0-headers.tar.gz
│ gyp http GET https://nodejs.org/download/release/v23.3.0/SHASUMS256.txt
│ gyp http GET https://nodejs.org/download/release/v23.3.0/win-x64/node.lib
│ gyp http 200 https://nodejs.org/download/release/v23.3.0/SHASUMS256.txt
│ gyp http 200 https://nodejs.org/download/release/v23.3.0/win-x64/node.lib
│ gyp ERR! find VS 
│ gyp ERR! find VS msvs_version not set from command line or npm config
│ gyp ERR! find VS VCINSTALLDIR not set, not running in VS Command Prompt
│ gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer, try re-running with '--loglevel silly' for more details.
│ gyp ERR! find VS 
│ gyp ERR! find VS Failure details: undefined
│ gyp ERR! find VS could not use PowerShell to find Visual Studio 2017 or newer, try re-running with '--loglevel silly' for more details.
│ gyp ERR! find VS 
│ gyp ERR! find VS Failure details: undefined
│ gyp ERR! find VS not looking for VS2017 as it is only supported up to Node.js 21
│ gyp ERR! find VS not looking for VS2017 as it is only supported up to Node.js 21
│ gyp ERR! find VS not looking for VS2017 as it is only supported up to Node.js 21
│ gyp ERR! find VS not looking for VS2015 as it is only supported up to Node.js 18
│ gyp ERR! find VS not looking for VS2013 as it is only supported up to Node.js 8
│ gyp ERR! find VS 
│ gyp ERR! find VS **************************************************************
│ gyp ERR! find VS You need to install the latest version of Visual Studio
│ gyp ERR! find VS including the "Desktop development with C++" workload.
│ gyp ERR! find VS For more information consult the documentation at:
│ gyp ERR! find VS https://github.com/nodejs/node-gyp#on-windows
│ gyp ERR! find VS **************************************************************
│ gyp ERR! find VS 
│ gyp ERR! configure error 
│ gyp ERR! stack Error: Could not find any Visual Studio installation to use
│ gyp ERR! stack at VisualStudioFinder.fail (D:\devlopTools\nvm\v23.3.0\node_global\node_modules\pnpm\dist\node_modules\node-gyp\lib\find-visualstudio.js:118:11)
│ gyp ERR! stack at VisualStudioFinder.findVisualStudio (D:\devlopTools\nvm\v23.3.0\node_global\node_modules\pnpm\dist\node_modules\node-gyp\lib\find-visualstudio.js:74:17)
│ gyp ERR! stack at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
│ gyp ERR! stack at async createBuildDir (D:\devlopTools\nvm\v23.3.0\node_global\node_modules\pnpm\dist\node_modules\node-gyp\lib\configure.js:112:18)
│ gyp ERR! stack at async run (D:\devlopTools\nvm\v23.3.0\node_global\node_modules\pnpm\dist\node_modules\node-gyp\bin\node-gyp.js:81:18)
│ gyp ERR! System Windows_NT 10.0.19045
│ gyp ERR! command "D:\\devlopTools\\nodejs\\node.exe" "D:\\devlopTools\\nvm\\v23.3.0\\node_global\\node_modules\\pnpm\\dist\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=D:\\develop\\eliza\\node_modules\\@discordjs\\opus\\prebuild\\node-v131-napi-v3-win32-x64-unknown-unknown\\opus.node" "--module_name=opus" …  
│ gyp ERR! cwd D:\develop\eliza\node_modules\@discordjs\opus
│ gyp ERR! node -v v23.3.0
│ gyp ERR! node-gyp -v v10.2.0
│ gyp ERR! not ok 
│ node-pre-gyp ERR! build error 
│ node-pre-gyp ERR! stack Error: Failed to execute 'D:\devlopTools\nodejs\node.exe D:\devlopTools\nvm\v23.3.0\node_global\node_modules\pnpm\dist\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\develop\eliza\node_modules\@discordjs\opus\prebuild\node-v131-napi-v3-win32-x64-unknown-unknown\opus.node --module_name=opus -…  
│ node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (D:\develop\eliza\node_modules\@discordjs\node-pre-gyp\lib\util\compile.js:85:20)
│ node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
│ node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1101:16)
│ node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:305:5)
│ node-pre-gyp ERR! System Windows_NT 10.0.19045
│ node-pre-gyp ERR! command "D:\\devlopTools\\nodejs\\node.exe" "D:\\develop\\eliza\\node_modules\\@discordjs\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
│ node-pre-gyp ERR! cwd D:\develop\eliza\node_modules\@discordjs\opus
│ node-pre-gyp ERR! node -v v23.3.0
│ node-pre-gyp ERR! node-pre-gyp -v v0.4.5
│ node-pre-gyp ERR! not ok 
│ Failed to execute 'D:\devlopTools\nodejs\node.exe D:\devlopTools\nvm\v23.3.0\node_global\node_modules\pnpm\dist\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\develop\eliza\node_modules\@discordjs\opus\prebuild\node-v131-napi-v3-win32-x64-unknown-unknown\opus.node --module_name=opus --module_path=D:\develop\eliza\n…  
└─ Failed in 7.8s at D:\develop\eliza\node_modules\@discordjs\opus
node_modules/@docusaurus/plugin-ideal-image/node_modules/sharp: Running install script, done in 4.8s
 ELIFECYCLE  Command failed with exit code 1.

**Comment by vnguyendc** [comment]
running into the same issue
